modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/RDBMSCoordinationStrategy.java

```
@@ -212,6 +212,23 @@ public void start(CoordinationConfigurableClusterAgent configurableClusterAgent,
 
         coordinatorElectionTask = new CoordinatorElectionTask();
         threadExecutor.execute(coordinatorElectionTask);
+
+        // Wait until node state become Candidate/Coordinator because thrift server needs to start after that.
+        int waitTime = 0;
+        int maxWaitTime = heartbeatMaxAge * 5;
+        while (currentNodeState == NodeState.ELECTION) {
+            try {
+                TimeUnit.MILLISECONDS.sleep(500);
+                waitTime = waitTime + 500;
+                if (waitTime == maxWaitTime) {
+                    throw new RuntimeException("Node status didn't change to Candidate/Coordinator within "
+                            + waitTime + " milliseconds.");
+                }
+            } catch (InterruptedException e) {
```

In this scenario, if interrupted I think we should throw an exception since most probably someone pressed Ctrl-c